torch_geometric/transforms/SKLearnPreprocessTransform.py

```diff
@@ -0,0 +1,40 @@
+from typing import List, Union
+
+import torch
+
+from torch_geometric.data import Data, HeteroData
+from torch_geometric.data.datapipes import functional_transform
+from torch_geometric.transforms import BaseTransform
+
+
+@functional_transform('sklearn_preprocess')
+class SKLearnPreprocessTransform(BaseTransform):
+    r"""Column-normalizes the attributes given in :obj:`attrs` to sum-up to standardize or scale them.
+
+    Mainly useful for edges, if the scalar for the collective data edges is stored
+    as data[(node1,to,node2)].edge_scaler = scaler
+
+    where the scaler can be StandardScaler or MinMaxScaler from scikit-learn that
+    has been fit to the total dataset
+
+    Args:
+        attrs (List[str]): The names of attributes to normalize.
+            (default: :obj:`["edge_attr"]`)
+    """
+    def __init__(self, attrs: List[str] = ["edge_attr"]):
+        self.attrs = attrs
```

Comment:
From `__call__` it looks like only edge attributes are transformed. Do we want to support modifying node attrs too.

Reply:
Yes, I think moving forward we should have support to scale `data[node].x` too. For this, i think we will need to consider all the `unique node features`. For example, if a Dataset contained 10 graphs in it, Dataset(10) , where each graph had ~100 nodes that were a sampled subset from a total collection 300 nodes , there will be repeats if we concatenate them all, to fit a scaler to it,  and some missing nodes that should be considered for min,std,max as well that could be held out in test. Not entirely clear on how to approach this yet. 

Once we have a scaler for each element of `data.metadata`(node_types and edge_types), we can transform individual graphs' nodes and edges accordingly.